dot/parachain/types/async_backing.go

```diff
@@ -17,3 +17,76 @@ type AsyncBackingParams struct {
 	// When async backing is disabled, the only valid value is 0.
 	AllowedAncestryLen uint32 `scale:"2"`
 }
+
+// InboundHrmpLimitations constraints on inbound HRMP channels.
+type InboundHrmpLimitations struct {
+	// An exhaustive set of all valid watermarks, sorted ascending.
+	//
+	// It's only expected to contain block numbers at which messages were
+	// previously sent to a para, excluding most recent head.
+	ValidWatermarks []BlockNumber `scale:"1"`
```

Comment:
I would just omit the `scale` struct tags in the introduced structures.  As long as the attributes are in the same order as the rust structure we don't need to explicitly add this struct tag.  I added the struct tag for cases where you see they override the attribute index with the codec macro (eg. `#[codec(index = 0)]`)